src/AdoNet/Orleans.Persistence.AdoNet/Storage/Provider/Orleans3CompatibleStringKeyHasher.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Text;
+
+namespace Orleans.Storage
+{
+    /// <summary>
+    /// Orleans v3-compatible hasher implementation for string-only grain key ids.
+    /// </summary>
+    internal class Orleans3CompatibleStringKeyHasher : IHasher
+    {
+        private readonly Orleans3CompatibleHasher _innerHasher;
+        private readonly string _grainType;
+
+        public Orleans3CompatibleStringKeyHasher(Orleans3CompatibleHasher innerHasher, string grainType)
+        {
+            _innerHasher = innerHasher;
+            _grainType = grainType;
+        }
+
+        /// <summary>
+        /// <see cref="IHasher.Description"/>
+        /// </summary>
+        public string Description { get; } = $"Orleans v3 hash function ({nameof(JenkinsHash)}).";
+
+        /// <summary>
+        /// <see cref="IHasher.Hash(byte[])"/>.
+        /// </summary>
+        public int Hash(byte[] data)
```

Comment:
```suggestion
        public int Hash(ReadOnlySpan<byte> data)
```
I'd go with the ROS on all these arguments.